monitoring/persistent_stats_history.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#include "monitoring/persistent_stats_history.h"
+
+#include <cstring>
+#include <string>
+#include <utility>
+#include "db/db_impl/db_impl.h"
+#include "port/likely.h"
+#include "util/string_util.h"
+
+namespace rocksdb {
+// 10 digit seconds timestamp => [Sep 9, 2001 ~ Nov 20, 2286]
+const int kNowSecondsStringLength = 10;
+const int kMicrosInSecond = 1000 * 1000;
+const std::string kFormatVersionKeyString =
+    "__persistent_stats_format_version__";
+const std::string kReaderVersionKeyString =
+    "__persistent_stats_reader_version__";
+const int kStatsCFCurrentFormatVersion = 1;
+const int kPersistentStatsReaderVersion = 1;
+
+std::string GetVersionKeyString(StatsVersionKeyType type) {
+  switch (type) {
+    case StatsVersionKeyType::kFormatVersion:
+      return kFormatVersionKeyString;
+    case StatsVersionKeyType::kReaderVersion:
+      return kReaderVersionKeyString;
+    default:
+      return "";
+  }
+}
+
+int DecodePersistentStatsVersionNumber(DBImpl* db, StatsVersionKeyType type) {
+  if (type >= StatsVersionKeyType::kKeyTypeMax) {
+    return -1;
+  }
+  std::string key = GetVersionKeyString(type);
+  ReadOptions options;
+  options.verify_checksums = true;
+  std::string result;
+  Status s = db->Get(options, db->PersistentStatsColumnFamily(), key, &result);
+  if (!s.ok() || result.length() == 0) {
```

Comment:
Do we really want to swallow the error? At least we should log in.

Reply:
changed return type to Status and have the caller log the errors